### DISCUSSION TITLE
Renovate ignoring PHP version when updating composer dependencies

### DISCUSSION THREAD

**Discussion by Jamesking56** [renovatebot/renovate#12297] [Request Help]
Hello,

It appears Renovate ignores the project's PHP version when updating dependencies.

I have an update PR in my private repo that is upgrading `laravel/framework` from `8.63.0` to `8.67.0`.

In this update, it also updates `psr/cache` from `1.0.1` to `3.0.0`.

However, `3.0.0` requires PHP 8 but my project is limited to PHP 7.4. My project correctly limits PHP version for dependencies in composer.json as follows:

```
...
"require": {
        "php": "^7.4",
...
```

But it appears Renovate is ignoring this and attempting to install dependencies that require PHP 8 anyway.

Is this a bug or configuration issue?

Link to the job log: https://app.renovatebot.com/dashboard#github/momentumstudioltd/cwmind-training-app
(Branch is `renovate/laravel-framework-8.x-lockfile`)

**Comment by Jamesking56** [comment]
Looks like this is caused by Renovate using `--ignore-platform-reqs` when updating Composer dependencies. This flag also ignores the set PHP requirement in `composer.json`.

**Comment by Jamesking56** [comment]
I have just configured `composerIgnorePlatformReqs` to `null` to see if it fixes the problem

**Comment by rarkins** [comment]
See https://github.com/renovatebot/renovate/issues/2355 but note it's still at status:requirements, meaning more input is needed before it is in the ready state for implementation
